src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1676,12 +1671,13 @@ public void setContextClassLoader(ClassLoader cl) {
      * release */
```

Comment:
Is it still desirable to use `ConcurrentReferenceHashMap` as suggested in comment? Or ClassValue is better choice?

Reply:
Hmm, no. The use of Reference here is to avoid strong binding to the Class key. However, we have seen weird interferences in ObjectStreamClass (see JDK-8277072) between this mechanism and class-unloading which prevents unloading the class sometimes (or at least, absent memory pressure that would trigger the GC to reclaim soft references). This code here in Thread.java is probably not affected by this problem, but ClassValue is still the cleaner way to establish a mapping between a Class and some value. I have removed the comment.